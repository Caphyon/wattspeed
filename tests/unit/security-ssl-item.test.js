import { shallowMount } from "@vue/test-utils";
import security_ssl_item from "../../src/components/panels/tech/items/security-ssl-item";

describe("security-ssl-item.vue", () => {
  it("Snapshot for no data", () => {
    const wrapper = shallowMount(security_ssl_item, {
      propsData: {
        message: '',
        type: '',
      },
    });

    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('div.scrollable--issue.alert--danger').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--warning').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--success').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--secondary').exists()).toBe(true);
    wrapper.destroy();
  });

  it("Snapshot for component level danger", () => {
    const wrapper = shallowMount(security_ssl_item, {
      propsData: {
        message: 'The connection to the server is not secure',
        type: 'error',
      },
    });

    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('div.scrollable--issue.alert--danger').exists()).toBe(true);
    expect(wrapper.find('div.scrollable--issue.alert--warning').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--success').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--secondary').exists()).toBe(false);
    wrapper.destroy();
  });

  it("Snapshot for component level warning", () => {
    const wrapper = shallowMount(security_ssl_item, {
      propsData: {
        message: 'This server is using weak cipher suites',
        type: 'warning',
      },
    });

    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('div.scrollable--issue.alert--danger').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--warning').exists()).toBe(true);
    expect(wrapper.find('div.scrollable--issue.alert--success').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--secondary').exists()).toBe(false);
    wrapper.destroy();
  });

  it("Snapshot for component level success", () => {
    const wrapper = shallowMount(security_ssl_item, {
      propsData: {
        message: 'Certificate is trusted',
        type: 'success',
      },
    });

    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('div.scrollable--issue.alert--danger').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--warning').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--success').exists()).toBe(true);
    expect(wrapper.find('div.scrollable--issue.alert--secondary').exists()).toBe(false);
    wrapper.destroy();
  });

});
