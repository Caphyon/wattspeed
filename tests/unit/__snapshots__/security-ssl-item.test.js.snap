// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`security-ssl-item.vue Snapshot for component level danger 1`] = `
<div
  class="scrollable--issue alert--danger"
>
  <p
    class="mb0 mt0"
  >
    The connection to the server is not secure
  </p>
</div>
`;

exports[`security-ssl-item.vue Snapshot for component level success 1`] = `
<div
  class="scrollable--issue alert--success"
>
  <p
    class="mb0 mt0"
  >
    Certificate is trusted
  </p>
</div>
`;

exports[`security-ssl-item.vue Snapshot for component level warning 1`] = `
<div
  class="scrollable--issue alert--warning"
>
  <p
    class="mb0 mt0"
  >
    This server is using weak cipher suites
  </p>
</div>
`;

exports[`security-ssl-item.vue Snapshot for no data 1`] = `
<div
  class="scrollable--issue alert--secondary"
>
  <p
    class="mb0 mt0"
  >
    
  </p>
</div>
`;
